addons/sale_project/models/project_project.py

```diff
@@ -383,10 +391,36 @@ def get_action(sol_id):
             action, res_id = action_per_sol.get(sol_id, (None, None))
             return {'action': {'name': action, 'resId': res_id, 'buttonContext': json.dumps({'active_id': sol_id, 'default_project_id': self.id})}} if action else {}
 
-        return [{
-            **sol_read,
-            **get_action(sol_read['id']),
-        } for sol_read in sols.with_context(with_price_unit=True).read(['display_name', 'product_uom_qty', 'qty_delivered', 'qty_invoiced', 'product_uom'])]
+        if section_id:
+            # The 'section_id' param is set, which means the method was called with the 'load more' button. We don't have to sort the result.
+            return [{
+                **sol_read,
+                **get_action(sol_read['id']),
+            } for sol_read in sols.with_context(with_price_unit=True).read(['name', 'product_uom_qty', 'qty_delivered', 'qty_invoiced', 'product_uom', 'product_id'])]
+
+        # The sols read on the initial load of the dashboard is done without any specific section. We have to sort the result in order to put the sol under their corresponding
+        # profitability section.
+        product_ids = {(sol_read['product_id'][0]) for sol_read in sols.read(['product_id'])}
+
+        products = self.env['product.product'].browse(product_ids)
+
+        section_id_per_product_id = {}
+        service_policy_to_invoice_type = self._get_service_policy_to_invoice_type()
+
+        for product in products:
+            if product.type != 'service':
+                section_id_per_product_id[product.id] = 'materials'
+            else:
+                invoice_type = service_policy_to_invoice_type[product.service_policy]
+                section_id_per_product_id[product.id] = invoice_type
+
+        items_id_per_section_id = self._get_items_id_per_section_id()
+
+        for sol_read in sols.with_context(with_price_unit=True).read(['name', 'product_uom_qty', 'qty_delivered', 'qty_invoiced', 'product_uom', 'product_id']):
```

Comment:
`_read_format` as well to avoid fetching again since we already have the data in cache thanks to the previous read.
```suggestion
        for sol_read in sols.with_context(with_price_unit=True)._read_format(['name', 'product_uom_qty', 'qty_delivered', 'qty_invoiced', 'product_uom', 'product_id']):
```